backend/drm/drm.c

```diff
@@ -1616,3 +1628,73 @@ void destroy_drm_connector(struct wlr_drm_connector *conn) {
 	wl_list_remove(&conn->link);
 	free(conn);
 }
+
+int wlr_drm_backend_create_lease(struct wlr_backend *backend,
+		struct wlr_output *output, uint32_t *lessee_id) {
+	assert(backend && wlr_backend_is_drm(backend));
+	assert(output && wlr_output_is_drm(output));
+
+	struct wlr_drm_backend *drm = get_drm_backend_from_backend(backend);
+
+	struct wlr_drm_connector *connector = (struct wlr_drm_connector *)output;
+	assert(connector->state != WLR_DRM_CONN_LEASED);
+
+	if (connector->backend != drm) {
+		wlr_log(WLR_ERROR, "output doesn't belong to the backend");
+		return -EINVAL;
+	}
+
+	uint32_t objects[] = {
+		connector->id,
+		connector->crtc->id,
```

Comment:
The connector is not guaranteed to have a CRTC. We should check that it has one.